Packs/Core/Integrations/CortexCoreIR/CortexCoreIR.py

```diff
@@ -173,6 +177,106 @@ def get_asset_details_command(client: Client, args: dict) -> CommandResults:
+    """
+    Create dict with all the command relevant data from result.
+
+    Args:
+        result (dict): Data from the execution of a command on a specific endpoint.
+
+    Returns:
+        dict: _description_
+    """
+    new_command = {'command': result['command'][1:]}
+    for key in COMMAND_DATA_KEYS:
+        new_command.update({key: result.get(key)})
+    return new_command
+
+
+def reformat_output(script_res: list) -> None:
+    """
+    Reformat the output so that each endpoint has its own result section, without any duplicated data.
+
+    Args:
+        script_res (list): The result from the polling command.
+    """
+    reformated_res: dict[str, Any] = {}
+    for response in script_res:
+        results = response.outputs.get('results')
+        for res in results:
+            endpoint_id = res.get('endpoint_id')
+            if endpoint_id in reformated_res:
+                reformated_res[endpoint_id]['executed_command'].append(new_executed_command(res))
+                reformated_res[endpoint_id].pop(res.get('command'))
+            else:
+                res['executed_command'] = [new_executed_command(res)]
+                res.pop(res.pop('command'))
+                for key in COMMAND_DATA_KEYS:
+                    res.pop(key)
+                reformated_res[endpoint_id] = res
+    new_res = [reformated_res[i] for i in reformated_res]
+    script_res[0].outputs['results'] = new_res
+
+
+def reformate_args(args: dict) -> None:
+    """
+    Reformat args before start polling command.
+
+    Args:
+        args (dict): Dictionary containing the arguments for the command.
+    """
+    commands = args.get('command')
+    if not commands:
+        raise DemistoException("'command' is a required argument.")
+    # the value of script_uid is the Unique identifier of execute_commands script.
+    args |= {'is_core': True, 'script_uid': 'a6f7683c8e217d85bd3c398f0d3fb6bf'}
+    is_raw_command = argToBoolean(args.get('is_raw_command', False))
+    commands_list = [commands] if is_raw_command else argToList(commands, args.get('command_separator'))
+    if args.get('command_type') == 'powershell':
+        commands_list = [form_powershell_command(command) for command in commands_list]
+    args['parameters'] = json.dumps({'commands_list': commands_list})
+
+
+def core_execute_command_command(client: Client, args: dict) -> PollResult:
+    """
+    Run executed_command script and reformat it's results.
+
+    Args:
+        client (Client): The client instance used to send the request.
+        args (dict): Dictionary containing the arguments for the command.
+
+    Returns:
+        PollResult: Reformated script_run_polling_command result.
+    """
+    reformate_args(args)
+    script_res = script_run_polling_command(args, client, statuses=('PENDING', 'IN_PROGRESS', 'PENDING_ABORT'))
+    if isinstance(script_res, list):
+        reformat_readable_output(script_res)
+        reformat_output(script_res)
+    return script_res
```

Comment:
It might be better to create a new command result object for the res and return it instead of editing the script_res in place.